src/main/webapp/app/js/goal-page.js

```diff
@@ -69,9 +72,67 @@ function addComment() {
                 document.getElementById("userComment").value = "";
                 onLoad();
             } else {
-                alert("Sompthing went wrong! error:404");
+                alert("Something went wrong! error:404");
                 return false;
             }
         });
     }
+}
+
+function isGoalEditable() {
+    fetch(path + "/api/goal/" + id + "/edit", {
+        "method": "GET",
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+    }).then(function (response) {
+        if (response.status === 200) {
+            return response.json();
+        } else {
+            alert("Something went wrong! error:404");
+            return false;
+        }
+    }).then(function (goalEdit) {
+        if (!goalEdit)
+            document.getElementById("edit-button").classList.add("w3-hide");
+        isMyGoal = goalEdit;
+    });
+}
+
+function editGoal() {
+    if (isMyGoal) {
+        document.getElementById("show-goal").classList.add("w3-hide");
+        document.getElementById("edit-goal").classList.remove("w3-hide");
+    } else {
+        alert("You can edit only your own goals");
+    }
+}
+
+function saveEditGoal() {
+    var goalTxt = document.getElementById("edit-goal-text");
+    var deadlineDate = document.getElementById("goal-deadline");
+    if (String(goalTxt.value) === "" || String(deadlineDate.value) === "") {
+        alert("You can not delete your goal!");
+        return false;
+    }
+    var dto = {
+        "goalMessage": goalTxt.value,
+        "deadline": deadlineDate.value
+    };
+    fetch(path + "/api/goal/" + id + "/edit", {
+        "method": "POST",
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(dto)
+    }).then(function (response) {
+        if (response.status === 204) {
+            redirectToGoalsAndComments(id);
+        } else {
+            alert("Something went wrong! error:404");
```

Comment:
Please append the actual `response.status` instead of `404`, or remove mention of status code.